.github/workflows/cpp_odbc.yml

```diff
@@ -0,0 +1,182 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: C++ ODBC
+
+on:
+  push:
+    branches:
+      - '**'
+      - '!dependabot/**'
+    tags:
+      - '**'
+    paths:
+      - '.github/workflows/cpp_odbc.yml'
+      - 'ci/scripts/cpp_*'
+      - 'cpp/src/arrow/flight/sql/odbc/*'
+  pull_request:
+    paths:
+      - '.github/workflows/cpp_odbc.yml'
+      - 'ci/scripts/cpp_*'
+      - 'cpp/src/arrow/flight/sql/odbc/*'
+  schedule:
+    - cron: '0 13 * * *'
```

Comment:
Do we need this?

Reply:
Since the workflow will only be run with ODBC (and ci cpp script) changes, making it run nightly will help catch any ODBC issues that occur, even when no ODBC changes have been merged into main.

Reply:
moving it as a job inside the cpp_extra.yaml as @kou pointed on another comment allows us to run this as part of the nightlies and also trigger it on demand via adding the `CI: extra` label so we can always run it as part of a job if we think it might be relevant.

Reply:
Thanks @raulcd and @kou for pointing me to `cpp_extra.yml`. I have tried to move the ODBC workflow to `cpp_extra.yml`. Currently it is building, will fix any build issues as I see them